seq_tools/validation/c160_file_r1_r2_check.py

```diff
@@ -0,0 +1,118 @@
+
+
+class Checker(BaseChecker):
+    def __init__(self, ctx, metadata):
+        super().__init__(ctx, metadata, __name__)
+
+    def check(self):
+        if not self.metadata.get('read_groups'):
+            message = "Missing 'read_groups' section in the metadata JSON"
+            self.logger.info(message)
+            self.message = message
+            self.status = 'INVALID'
+            return
+ 
+        if not self.metadata.get('files'):
+            message = "Missing 'files' section in the metadata JSON"
+            self.logger.info(message)
+            self.message = message
+            self.status = 'INVALID'
+            return
+
+        fns = set()
+        for rg in self.metadata.get('read_groups'):
+            if 'is_paired_end' not in rg:
+                message = "Required field 'is_paired_end' is not found in metadata JSON"
+                self.logger.info(message)
+                self.message = message
+                self.status = 'INVALID'
+                return
+
+            if not isinstance(rg['is_paired_end'], bool):
+                message = "Required field 'is_paired_end' should be Boolean type."
+                self.logger.info(message)
+                self.message = message
+                self.status = 'INVALID'
+                return
+            
+            if rg['is_paired_end']:
+                if not 'file_r1' in rg or not rg['file_r1']:
+                    message = "Required field 'file_r1' is not found in metadata JSON for paired end sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                    return
+                if not 'file_r2' in rg or not rg['file_r2']:
+                    message = "Required field 'file_r2' is not found in metadata JSON for paired end sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                    return
+                if rg['file_r1'].endswith('.bam') and not rg['file_r1'] == rg['file_r2']:
+                    message = "Fields 'file_r1' and 'file_r2' should be the same for paired end BAM sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                if not rg['file_r1'].endswith('.bam') and rg['file_r1'] == rg['file_r2']:
+                    message = "Fields 'file_r1' and 'file_r2' should NOT be the same for paired end FASTQ sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                
+            else:
+                if 'file_r2' in rg and rg['file_r2']:
+                    message = "Field 'file_r2' should NOT been populated in metadata JSON for single end sequencing reads."
```

Comment:
consider:  `"Field 'file_r2' must be 'null' in metadata JSON for single end sequencing in read group: %s." % rg['submitter_read_group_id']`